analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.Enumerable.cs

```diff
@@ -69,25 +69,64 @@ internal sealed partial class Invocation
         "UnionBy",
         nameof(Enumerable.Where),
         nameof(Enumerable.Zip),
-    };
+    ];
+
+    private static readonly HashSet<string> ElemenetExistsCheckMethods =
+    [
+        nameof(Enumerable.Contains),
+        nameof(Enumerable.Any),
+        nameof(List<int>.Exists)
+    ];
 
     private static ProgramState[] ProcessLinqEnumerableAndQueryable(ProgramState state, IInvocationOperationWrapper invocation)
     {
         var name = invocation.TargetMethod.Name;
+        var states = ProcessBoolCollectionMethods(state, invocation);
         if (ReturningNotNull.Contains(name))
         {
-            return state.SetOperationConstraint(invocation, ObjectConstraint.NotNull).ToArray();
+            return states.Select(x => x.SetOperationConstraint(invocation, ObjectConstraint.NotNull)).ToArray();
```

Comment:
I think not really because in the else if checking about the firstordefault etc I'll need to check the empty/notempty set by the ProcessElementExists method